velox/exec/OutputBuffer.h

```diff
@@ -124,10 +154,35 @@ class DestinationBuffer {
   // The sequence number of the first item to pass to 'notify'.
   int64_t notifySequence_{0};
   uint64_t notifyMaxBytes_{0};
+  DestinationBufferStats stats_;
 };
 
 class Task;
 
+struct OutputBufferStats {
+  OutputBufferStats(
+      core::PartitionedOutputNode::Kind k,
+      bool canAddBuffer,
+      bool canAddPage,
+      bool finish,
+      const std::vector<DestinationBufferStats>& buffers)
+      : kind(k),
+        canAddBuffers(canAddBuffer),
+        canAddPages(canAddPage),
+        finished(finish),
+        destinationBuffers(buffers) {}
+
+  const core::PartitionedOutputNode::Kind kind;
+
+  /// States of the OutputBuffer.
+  const bool canAddBuffers;
```

Comment:
It might be hard to connect these flags to OutputBuffer API. Perhaps, rename to noMoreBuffers and noMoreData. These would match OutputBuffer::noMoreData() API and noMoreBuffers flags passed to OutputBuffer::updateBuffers.

Reply:
Make sense, modified.